Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -2977,13 +2969,15 @@ extension BrowserViewController: KeyboardHelperDelegate {
         keyboardState = nil
         updateViewConstraints()
         //If the searchEngineButton exists remove it form the keyboard
-        if self.customSearchEngineButton.superview != nil {
-            self.customSearchEngineButton.removeFromSuperview()
+        if #available(iOS 9.0, *) {
+            if let buttonGroup = customSearchBarButton?.buttonGroup  {
+                buttonGroup.barButtonItems = buttonGroup.barButtonItems.filter { $0.title != "SearchAdd" }
```

Comment:
Why do you need the title for this rather than just `$0 != customSearchBarButton`?